geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLRules.scala

```diff
@@ -113,8 +114,78 @@ object SQLRules extends LazyLogging {
       }
     }
 
+    private def extractGridId(envelopes: List[Envelope], e: org.apache.spark.sql.catalyst.expressions.Expression): Option[List[Int]] = e match {
+      case And(l, r) => extractGridId(envelopes, l).orElse(extractGridId(envelopes, r))
```

Comment:
Ah you're right. If there are multiple predicates joined with `And`, then only the first one will be used. This would return the wrong results in some cases. 
I can change it to union the two grid ids instead and we wouldn't have this problem.